tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9025,33 +9010,106 @@ TEST_F(ExecutionTest, BarycentricsTest) {
     XMFLOAT2 p0(pTriangleData[0], pTriangleData[1]);
     XMFLOAT2 p1(pTriangleData[triangleVertexSizeInFloat], pTriangleData[triangleVertexSizeInFloat + 1]);
     XMFLOAT2 p2(pTriangleData[triangleVertexSizeInFloat * 2], pTriangleData[triangleVertexSizeInFloat * 2 + 1]);
-
+    
+    // Seems like the 3 floats must add up to 1 to get accurate results.
     XMFLOAT3 barycentricWeights[4] = {
-        XMFLOAT3(0.3333f, 0.3333f, 0.3333f),
+        XMFLOAT3(0.4f, 0.2f, 0.4f),
         XMFLOAT3(0.5f, 0.25f, 0.25f),
         XMFLOAT3(0.25f, 0.5f, 0.25f),
         XMFLOAT3(0.25f, 0.25f, 0.50f)
-    };
+    };    
 
-    float tolerance = 0.001f;
+    float tolerance = 0.02f;
     for (unsigned i = 0; i < sizeof(barycentricWeights) / sizeof(XMFLOAT3); ++i) {
         float w0 = barycentricWeights[i].x;
         float w1 = barycentricWeights[i].y;
         float w2 = barycentricWeights[i].z;
         float x1 = w0 * p0.x + w1 * p1.x + w2 * p2.x;
         float y1 = w0 * p0.y + w1 * p1.y + w2 * p2.y;
         // map from x1 y1 to rtv pixels
-        int pixelX = (int)((x1 + 1) * (width - 1) / 2);
-        int pixelY = (int)((1 - y1) * (height - 1) / 2);
+        int pixelX = (int)round((x1 + 1) * (width - 1) / 2);
+        int pixelY = (int)round((1 - y1) * (height - 1) / 2); //1.3332 * 63 = 83.999, /2 = 41.999 truncated = 41
```

Comment:
This is going to truncate before it rounds. You'll need to change the `/ 2` to `/ 2.0` or `/ 2f`

I think the comment is obsolete.

Reply:
Tex points out that the single floats in this equation would have made the whole of it floating point. Even so, I like to have the divisor make which kind of division is intended clear.